ui/v2.5/src/components/List/Filters/RatingFilter.tsx

```diff
@@ -1,135 +1,71 @@
-          type="number"
-          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-            onChanged(e, "value")
-          }
-          onBlur={onBlurInput}
-          defaultValue={
-            convertToRatingFormat(criterion.value?.value, ratingSystem) ?? ""
-          }
-          placeholder={intl.formatMessage({ id: "criterion.value" })}
+    return (
+      <div>
+        <RatingSystem
+          value={criterion.value[field]}
+          onSetRating={(value) => {
+            onValueChanged({
+              ...criterion.value,
+              [field]: value ?? defaultValue,
+            });
+          }}
+          valueRequired
         />
-      </Form.Group>
+      </div>
     );
   }
 
-  let lowerControl: JSX.Element | null = null;
   if (
+    criterion.modifier === CriterionModifier.Equals ||
+    criterion.modifier === CriterionModifier.NotEquals ||
     criterion.modifier === CriterionModifier.GreaterThan ||
-    criterion.modifier === CriterionModifier.Between ||
-    criterion.modifier === CriterionModifier.NotBetween
+    criterion.modifier === CriterionModifier.LessThan
   ) {
-    lowerControl = (
-      <Form.Group>
-        <Form.Control
-          className="btn-secondary"
-          type="number"
-          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-            onChanged(e, "value")
-          }
-          onBlur={onBlurInput}
-          defaultValue={
-            convertToRatingFormat(criterion.value?.value, ratingSystem) ?? ""
-          }
-          placeholder={intl.formatMessage({ id: "criterion.greater_than" })}
-        />
-      </Form.Group>
-    );
+    // ensure value is set
+    if (!criterion.value.value) {
+      onValueChanged({ value: 100, value2: undefined });
+    }
+
+    return getRatingSystem("value");
   }
 
-  let upperControl: JSX.Element | null = null;
   if (
-    criterion.modifier === CriterionModifier.LessThan ||
     criterion.modifier === CriterionModifier.Between ||
     criterion.modifier === CriterionModifier.NotBetween
   ) {
-    upperControl = (
-      <Form.Group>
-        <Form.Control
-          className="btn-secondary"
-          type="number"
-          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-            onChanged(
-              e,
-              criterion.modifier === CriterionModifier.LessThan
-                ? "value"
-                : "value2"
-            )
-          }
-          onBlur={onBlurInput}
-          defaultValue={
-            convertToRatingFormat(
-              criterion.modifier === CriterionModifier.LessThan
-                ? criterion.value?.value
-                : criterion.value?.value2,
-              ratingSystem
-            ) ?? ""
-          }
-          placeholder={intl.formatMessage({ id: "criterion.less_than" })}
-        />
-      </Form.Group>
+    // ensure values are set
+    if (!criterion.value.value || !criterion.value.value2) {
+      onValueChanged({ value: 20, value2: 100 });
```

Comment:
Is there a reason we aren't starting from 0-100? Decrementing from 0.1 resets to 2.0.

Reply:
Good pickup. I've removed the default value code. Instead, I've made it so that the `Add/Update Filter` button is disabled if the criterion is invalid - rating not set or between range is invalid.